src/Symfony/Component/Serializer/Exception/NotNormalizableValueException.php

```diff
@@ -16,4 +16,46 @@
  */
 class NotNormalizableValueException extends UnexpectedValueException
 {
+    private $currentType = null;
+    private $expectedTypes = null;
+    private $path = null;
+    private $useMessageForUser = null;
+
+    /**
+     * @param bool $useMessageForUser If the message passed to this exception is something that can be shown
```

Comment:
WDYT of `InvalidTypeException/InvalidValueException extends NotNormalizableValueException`? i tend to believe it makes semantics a bit more straightforward :)

Reply:
Why not, but this is not related to the useMessageForUser right ?

Reply:
IIUC currently any useMessageForUser=true case, is actually a InvalidTypeException/InvalidValueException case

reasonably i'd say

catch(InvalidTypeException/InvalidValueException) => 400 + user message (eg. useMessageForUser=true)
catch(NotNormalizableValueException) => 400 + safe message (eg. useMessageForUser=false)

Reply:
the caught exception could be a compound one, to further digest. Or as you obtain them currenly, by ref. Then the context could be determined using instanceof.